support-lib/CMakeLists.txt

```diff
@@ -0,0 +1,76 @@
+cmake_minimum_required(VERSION 3.6.0)
+
+project(Djinni)
+
+include(GNUInstallDirs)
+
+set(SRC_SHARED
+  "djinni_common.hpp"
+  "proxy_cache_interface.hpp"
+  "proxy_cache_impl.hpp"
+)
+
+set(SRC_JNI
+  "jni/djinni_support.hpp"
+  "jni/Marshal.hpp"
+  "jni/djinni_support.cpp"
+)
+
+set(SRC_OBJC
+  "objc/DJICppWrapperCache+Private.h"
+  "objc/DJIError.h"
+  "objc/DJIMarshal+Private.h"
+  "objc/DJIObjcWrapperCache+Private.h"
+  "objc/DJIError.mm"
+  "objc/DJIProxyCaches.mm"
+)
+
+option(DJINNI_STATIC_LIB "Build Djinni as a static library instead of dynamic (the default)." off)
+if(DJINNI_STATIC_LIB)
+  add_library(djinni STATIC ${SRC_SHARED})
+else()
+  add_library(djinni SHARED ${SRC_SHARED})
+endif()
+source_group("" FILES ${SRC_SHARED})
+
+set_target_properties(djinni PROPERTIES
+  CXX_STANDARD 11
```

Comment:
Any way to make this a minimum requirement rather than an absolute one?
I worry a bit about users' wanting their full codebase (including Djinni-generated files, and the support library) to be built with the same standard version, which could be >11